File: app/src/main/java/com/enipro/data/remote/model/PremiumDetails.java
package com.enipro.data.remote.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.ParcelConstructor;
import org.parceler.ParcelProperty;

@org.parceler.Parcel
public class PremiumDetails {


    @SerializedName("accountNumber")
    @Expose
    public String accountNumber;

    @SerializedName("bank")
    @Expose
    public String bank;

    @SerializedName("payment_amount")
    @Expose
    public int payment_amount;

    @SerializedName("payment_code")
    @Expose
    public String payment_code;

    @ParcelConstructor
    public PremiumDetails(String accountNumber, String bank, @ParcelProperty("amount") int amount) {
        this.accountNumber = accountNumber;
        this.bank = bank;
        this.payment_amount = amount;
    }

    public String getAccountNumber() {
        return accountNumber;
    }

    public void setAccountNumber(String accountNumber) {
        this.accountNumber = accountNumber;
    }

    public String getBank() {
        return bank;
    }

    public void setBank(String bank) {
        this.bank = bank;
    }

    @ParcelProperty("amount")
    public int getPayment_amount() {
        return payment_amount;
    }

    public void setPayment_amount(int payment_amount) {
        this.payment_amount = payment_amount;
    }

    public String getPayment_code() {
        return payment_code;
    }

    public void setPayment_code(String payment_code) {
        this.payment_code = payment_code;
    }

}

File: app/src/main/java/com/enipro/data/remote/model/PremiumDetails.kt
package com.enipro.data.remote.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class PremiumDetails
constructor(@field:SerializedName("accountNumber")
            @field:Expose
            var accountNumber: String,
            @field:SerializedName("bank")
            @field:Expose
            var bank: String,
            @field:SerializedName("payment_amount")
            @field:Expose
            var payment_amount: Int) : Parcelable {

    @SerializedName("payment_code")
    @Expose
    var payment_code: String? = null

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readInt()) {
        payment_code = parcel.readString()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(accountNumber)
        parcel.writeString(bank)
        parcel.writeInt(payment_amount)
        parcel.writeString(payment_code)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<PremiumDetails> {
        override fun createFromParcel(parcel: Parcel): PremiumDetails {
            return PremiumDetails(parcel)
        }

        override fun newArray(size: Int): Array<PremiumDetails?> {
            return arrayOfNulls(size)
        }
    }

}
